code/fripp.c

```diff
@@ -0,0 +1,139 @@
+/************************************************************\
+ * Copyright 2014 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#if HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <flux/core.h>
+#include <unistd.h>
+#include <stdbool.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <sys/socket.h>
+
+#define FRIPP_MAX_PACKET_LEN 1440
+#define FRIPP_INTERNAL_BUF_SIZE (FRIPP_MAX_PACKET_LEN * 5)
+
+struct fripp_ctx {
+    flux_t *h;
+    struct sockaddr_in si_server;
+};
+
+/* Bind a socket address to hostname:port.
+ */
+int fripp_bind_server (struct fripp_ctx *ctx, const char *hostname, uint16_t port)
+{
+    memset (&ctx->si_server, 0, sizeof (ctx->si_server));
+    ctx->si_server.sin_family = AF_INET;
+    ctx->si_server.sin_port = htons (port);
+
+    if (!inet_aton (hostname, &ctx->si_server.sin_addr)) {
+        flux_log_error (ctx->h, "error creating server address");
+        return -1;
+    }
+
+    return 0;
+}
+
+/* Split a packet that is longer than FRIPP_MAX_PACKET_LEN
+ * by the last '\n' character, and replace it with a '\0'.
+ */
+static int split_packet (char *packet)
+{
+    for (int i = FRIPP_MAX_PACKET_LEN - 1; i >= 0; i--) {
+        if (packet[i] == '\n' || packet[i] == '\0') {
+            packet[i] = '\0';
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+
+/* Send the 'packet' string in one or more udp packets to
+ * the bound server within 'ctx'.
+ */
+static int fripp_send_metrics (struct fripp_ctx *ctx, char *packet)
+{
+    int sock, sock_len = sizeof (ctx->si_server);
+
+    if ((sock = socket(AF_INET, SOCK_DGRAM, 0)) == -1)
+        return -1;
+
+    int len = strlen (packet);
+    bool split = false;
+
+    if (len > FRIPP_MAX_PACKET_LEN) {
+        split = true;
+
+        if ((len = split_packet (packet)) == -1) {
+            close (sock);
+            return -1;
+        }
+    }
+
+    if (sendto (sock, packet, len, 0, (void *) &ctx->si_server, sock_len) < 0)
+        flux_log (ctx->h, LOG_ALERT, "packet %s dropped", packet);
+
+    close (sock);
+
+    if (split)
+        return fripp_send_metrics (ctx, &packet[len + 1]);
```

Comment:
Instead of recursion, it might be a little nicer to do this iteratively.  For example, find the largest coherent chunk that can be sent, move pointer to last char + 1, repeat....